nuxeo-admin-console-web/angular-app/src/app/shared/services/network.service.ts

```diff
@@ -10,19 +13,30 @@ type EndpointName = keyof typeof REST_END_POINTS;
   providedIn: "root",
 })
 export class NetworkService {
-  constructor(private http: HttpClient,private nuxeoJsClientService: NuxeoJSClientService) {}
+  constructor(
+    private http: HttpClient,
+    private nuxeoJsClientService: NuxeoJSClientService
+  ) {}
 
   getAPIEndpoint = (name: EndpointName): string => {
     const config = REST_END_POINT_CONFIG[name];
     return `${this.nuxeoJsClientService.getApiUrl()}${config.endpoint}`;
   };
 
-  makeHttpRequest<T>(endpointName: EndpointName, data?: Record<string, unknown>): Observable<T> {
+  makeHttpRequest<T>(
+    endpointName: EndpointName,
+    data?: Record<string, unknown>
+  ): Observable<T> {
     const config = REST_END_POINT_CONFIG[endpointName];
     let url = this.getAPIEndpoint(endpointName);
     const method = config.method || "PUT";
     let params = new HttpParams();
 
+    if (data?.["urlParam"] && url.indexOf("{id}") > -1) {
```

Comment:
This logic should be more generic to allow any other url param besides id.
Logic -
- Check if there is a urlParam
- If yes, then extract the object from urlParam value and for each key, check if the place holder lies in the url
For eg: if urlParam has "abc" as a key, then check if apiurl/{abc}/xyz is there in the url. Similartly if urlParam has "id" as a key, then check if apiurl/{id} is there.
- If yes, then replace {id} with the actual value from the object